spyder/plugins/application/widgets/install.py

```diff
@@ -0,0 +1,289 @@
+# -*- coding: utf-8 -*-
+
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""Update installation widgets."""
+
+# Standard library imports
+import logging
+import os
+import subprocess
+import tempfile
+import threading
+from urllib.request import urlretrieve
+
+# Third-party imports
+from qtpy.QtCore import Qt, Signal
+from qtpy.QtWidgets import (QDialog, QHBoxLayout, QMessageBox,
+                            QLabel, QProgressBar, QPushButton, QVBoxLayout,
+                            QWidget)
+
+# Local imports
+from spyder import __version__
+from spyder.api.translations import get_translation
+from spyder.utils.icon_manager import ima
+from spyder.utils.programs import is_module_installed
+
+logger = logging.getLogger(__name__)
+
+# Localization
+_ = get_translation('spyder')
+
+# Update installation process statuses
+NO_STATUS = __version__
+DOWNLOADING_INSTALLER = _("Downloading update")
+INSTALLING = _("Installing update")
+FINISHED = _("Installation finished")
+PENDING = _("Update available")
+CHECKING = _("Checking for updates")
+CANCELLED = _("Cancelled update")
+
+INSTALL_INFO_MESSAGES = {
+    DOWNLOADING_INSTALLER: _("Downloading latest Spyder update"),
+    INSTALLING: _("Installing Spyder update"),
+    FINISHED: _("Spyder update installation finished"),
+    PENDING: _("Spyder update available to download"),
+    CHECKING: _("Checking for Spyder updates"),
+    CANCELLED: _("Spyder update cancelled")
```

Comment:
```suggestion
    DOWNLOADING_INSTALLER: _("Downloading Spyder {version}"),
    INSTALLING: _("Installing Spyder {version}"),
    FINISHED: _("Finished installing Spyder {version}"),
    PENDING: _("Spyder {version} available to download"),
    CHECKING: _("Checking for new Spyder version"),
    CANCELLED: _("Spyder update cancelled")
```

Will need changes (something like `message.format(version=version)` where this is used) but as discussed in our meeting, this will avoid redundancy with the other message and provide useful information.